sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ClientSideRequestStatistics.java

```diff
@@ -168,63 +183,25 @@ void recordAddressResolutionEnd(String identifier) {
 
     @Override
     public String toString() {
-        StringBuilder stringBuilder = new StringBuilder();
-
         //  need to lock in case of concurrent operations. this should be extremely rare since toString()
         //  should only be called at the end of request.
         synchronized (this) {
-
-            //  first trace request start time, as well as total non-head/headfeed requests made.
-            stringBuilder.append("RequestStartTime: ")
-                .append("\"").append(this.requestStartTime.format(RESPONSE_TIME_FORMATTER)).append("\"")
-                .append(", ")
-                .append("RequestEndTime: ")
-                .append("\"").append(this.requestEndTime.format(RESPONSE_TIME_FORMATTER)).append("\"")
-                .append(", ")
-                .append("Duration: ")
-                .append(Duration.between(requestStartTime, requestEndTime).toMillis())
-                .append(" ms, ")
-                .append("Connection Mode : " + this.connectionMode.toString() + ", ")
-                .append("NUMBER of regions attempted: ")
-                .append(this.regionsContacted.isEmpty() ? 1 : this.regionsContacted.size())
-                .append(System.lineSeparator());
-
-            //  take all responses here - this should be limited in number and each one contains relevant information.
-            for (StoreResponseStatistics storeResponseStatistics : this.responseStatisticsList) {
-                stringBuilder.append(storeResponseStatistics.toString()).append(System.lineSeparator());
-            }
-
-            //  take all responses here - this should be limited in number and each one is important.
-            for (AddressResolutionStatistics value : this.addressResolutionStatistics.values()) {
-                stringBuilder.append(value.toString()).append(System.lineSeparator());
-            }
-
+            requestLatency= getRequestLatency().toMillis();
```

Comment:
toString() should not change the state of the object. This is somewhat an anti-pattern

Reply:
Are we talking about supplementalResponseStatisticsList , where we chopping off the top 10 items? I am aware of it , did it on purpose , no logic issue in this particular case, however if we want to avoid it , then we need to come up with probably another list , which will add overhead. Lets discuss offline for better solution

Reply:
Moved logic out of toString() of clientSideRequestDiagnostics , and using separate serializer